frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadFileForm.component.tsx

```diff
@@ -167,169 +148,112 @@ export const UploadFileForm = ({
+				teamspace,
+				project,
+				presetContainer._id,
+			);
+		}
 		FederationsActionsDispatchers.fetchFederations(teamspace, project);
 	}, []);
 
 	return (
 		<FormProvider {...formData}>
 			<Modal
 				open={open}
-				onSubmit={handleSubmit(onSubmit)}
+				onSubmit={!isUploading ? onSubmit : finishedSubmit}
 				onClickClose={onClickClose}
-				onKeyPress={(e) => e.key === 'Enter' && e.preventDefault()}
+				onKeyPress={preventEnter}
 				maxWidth="xl"
-				isValid={(isValid && !fileError && !isUploading) || (isUploading && allUploadsComplete)}
+				isValid={!isUploading ? isValid : allUploadsComplete}
 				contrastColorHeader
+				fields={fields}
+				isUploading={isUploading}
 				{...uploadModalLabels({ isUploading, fileCount: fields.length })}
 			>
-				<UploadsContainer>
-					<UploadsListScroll>
-						<Padding>
-							{!!fields.length && (
-								<>
-									<UploadsListHeader
-										onSortingChange={setSortConfig}
-										defaultSortConfig={DEFAULT_SORT_CONFIG}
-									>
-										<DashboardListHeaderLabel key="file" name="file.name" minWidth={122}>
-											<FormattedMessage id="uploads.list.header.filename" defaultMessage="Filename" />
-										</DashboardListHeaderLabel>
-										<DashboardListHeaderLabel key="destination" width={352}>
-											<FormattedMessage id="uploads.list.header.destination" defaultMessage="Destination" />
-										</DashboardListHeaderLabel>
-										<DashboardListHeaderLabel key="revisionName" width={isUploading ? 359 : 399}>
-											<FormattedMessage id="uploads.list.header.revisionName" defaultMessage="Revision Name" />
-										</DashboardListHeaderLabel>
-										<DashboardListHeaderLabel key="progress" width={297} hidden={!isUploading}>
-											<FormattedMessage id="uploads.list.header.progress" defaultMessage="Upload Progress" />
-										</DashboardListHeaderLabel>
-									</UploadsListHeader>
+				<UploadFileFormContextComponent fields={fields}>
+					<UploadsContainer>
+						<UploadsListScroll>
+							<Padding>
+								{!!fields.length && (
 									<UploadList
-										values={sortedList}
-										selectedIndex={getSortedListSelectedIndex()}
+										values={reduceFileData(fields)}
 										isUploading={isUploading}
-										onClickEdit={setSelectedIndex}
-										onClickDelete={onClickDelete}
-										getOriginalIndex={getOriginalIndex}
+										removeUploadById={removeUploadById}
 									/>
-								</>
-							)}
-							<DropZone
-								hidden={isUploading}
-								onDrop={addFilesToList}
-								accept={getSupportedFileExtensions()}
-							>
-								<Typography variant="h3" color="secondary">
-									<FormattedMessage id="dragAndDrop.drop" defaultMessage="Drop files here" />
-								</Typography>
-								<Typography variant="h5" color="secondary">
-									<FormattedMessage id="dragAndDrop.or" defaultMessage="or" />
-								</Typography>
-								<FileInputField
+								)}
+								<DropZone
```

Comment:
Drag and drop crashes
![image](https://github.com/3drepo/3drepo.io/assets/62481250/91375960-15d6-4622-9a2a-2b8d216a510f)